qiskit/visualization/circuit/matplotlib.py

```diff
@@ -64,6 +65,24 @@ class MatplotlibDrawer:
 
     _mathmode_regex = re.compile(r"(?<!\\)\$(.*)(?<!\\)\$")
 
+    @deprecate_arg("gregs", since="0.20.0")
+    @deprecate_arg("cregs", since="0.20.0")
+    @deprecate_arg("layout", since="0.20.0")
+    @deprecate_arg("global_phase", since="0.20.0")
+    @deprecate_arg("calibrations", since="0.20.0")
+    @deprecate_arg(
+        "circuit",
+        deprecation_description=(
+            "Not setting the ``circuit`` argument in the ``MatplotlibDrawer`` constructor`"
+        ),
+        since="0.20.0",
+        predicate=lambda circuit: circuit is None,
+        additional_msg=(
+            "The ``circuit`` argument must be a valid QuantumCircuit. If ``circuit`` is set to "
+            "``None``, then the circuit will be built using the ``qubits`` and ``clbits`` "
+            "arguments for rendering the drawing."
+        ),
+    )
```

Comment:
Similar comment to above about this being technically possible with the deprecators, but probably ending up more confusing for users than just have the manual warning with completely custom wording.